hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/BucketEndpoint.java

```diff
@@ -264,12 +267,15 @@ public Response get(
       response.setTruncated(false);
     }
 
-    AUDIT.logReadSuccess(buildAuditMessageForSuccess(s3GAction,
-        getAuditParameters()));
     int keyCount =
         response.getCommonPrefixes().size() + response.getContents().size();
-    getMetrics().updateGetBucketSuccessStats(startNanos);
+    long opLatencyNs =
+        getMetrics().updateGetBucketSuccessStats(startNanos);
     getMetrics().incListKeyCount(keyCount);
+    perf.appendCount(keyCount);
+    perf.appendOpLatencyNanos(opLatencyNs);
+    AUDIT.logReadSuccess(buildAuditMessageForSuccess(s3GAction,
```

Comment:
`AUDIT` may not log reads because of performance, but the `perf` does always `build()`. Should the performance data be passed as a lamba instead, so that a new string is not actually built when read audi logs are not enabled?

Reply:
This is a problem that deserves attention
We currently generate `AuditMessages` via `buildAuditMessageForSuccess`, etc. In the `AuditMessageForSuccess` we perform the `build` of all relevant messages, regardless of whether or not the logs are logged. 
This is probably not just a perf problem.

Reply:
I agree that the problem also applies to the existing, "functional" part of the audit message.

Audit logging is enabled by default, and we need several changes to improve it, so I think we should do it separately.

Reply:
Reopened HDDS-2638 for the improvement of audit message processsing/allocations.